docs: add introduction for daffodil (#3552)
graycoreio/daffodil: docs/guides/introduction/about.md

# Introduction

## What is Daffodil?

> Daffodil is a Typescript framework for building headless ecommerce stores. It is built on top of Angular and provides a suite of composable packages that allow you to build complex ecommerce stores.

Reviewer: ```suggestion
Daffodil is a frontend framework for building headless ecommerce stores. It is built on top of Angular and provides a suite of composable packages that allow you to build complex ecommerce stores.
```

> Daffodil is a Typescript framework for building headless ecommerce stores. It is built on top of Angular and provides a suite of composable packages that allow you to build complicated ecommerce stores.

Reviewer: `complex` instead of complicated?
```suggestion
Daffodil is a Typescript framework for building headless ecommerce stores. It is built on top of Angular and provides a suite of composable packages that allow you to build complicated ecommerce stores.
```
```suggestion
Daffodil is an Angular framework for building headless ecommerce stores. It provides a suite of composable packages to help you build complex ecommerce solutions.
```

Typescript isn't a framework?

Repeating `ecommerce stores` back to back sounds redundant.

Author: Daffodil is a Typescript framework. Typescript is modifying the word framework, we're not saying that Typescript is a framework. I could clarify that by saying that "Daffodil is a frontend framework, written in Typescript, for building ecommerce stores."

Reviewer: > Daffodil is a Typescript framework. Typescript is modifying the word framework, we're not saying that Typescript is a framework. I could clarify that by saying that "Daffodil is a frontend framework, written in Typescript, for building ecommerce stores."

I think "Daffodil is a frontend Angular framework for building ecommerce stores" is good enough unless we trying to stray away from Angular. I don't think mentioning that it's written in typescript serves any important purpose as I assume that's the standard for most frontend frameworks nowadays?

Author: I fixed 

> complex instead of complicated? 

in 11150ae81760bd488e22fefae055aded7c31f8a7

> Daffodil is a frontend Angular framework for building ecommerce stores

Angular itself is a framework, so this phrasing sounds strange. It think this would be accurate "Daffodil is a frontend framework for building ecommerce stores".

> These packages, in turn, provide tools and architectures that allow you to quickly build storefronts that are:

Reviewer: ```suggestion
These packages provide tools and architectures that allow you to quickly build storefronts that are:
```

1. Performant
2. Reliable
3. Interoperable
4. Maintainable
5. Accessible

> ## The Framework

Reviewer: ```suggestion
## The framework
```

I have been sentence casing headings

Daffodil is built with flexibility and scalability in mind. Here are a few features that we believe make Daffodil special:

> - **Drivers**: [Drivers](/docs/guides/drivers.md) are Daffodil's lifeblood. The drivers enable you to easily integrate with a variety of ecommerce backends, payment systems, and other services, allowing you to extend functionality without disrupting the core framework.

Reviewer: ```suggestion
- **Drivers**: [Drivers](/docs/guides/drivers.md) are Daffodil's lifeblood. The drivers enable you to easily integrate with various ecommerce backends, payment systems, and other services, allowing you to extend functionality without disrupting the core framework.
```

> - **Drivers** [Drivers](/docs/guides/drivers.md) are Daffodil's lifeblood. The drivers enable you to easily integrate with a variety of ecommerce backends, payment systems, and other services, allowing you to extend functionality without disrupting the core framework.

Reviewer: Minor formatting change:

```
- **Drivers**: [Drivers](/docs/guides/drivers.md) are Daffodil's lifeblood. The drivers enable you to easily integrate with a variety of ecommerce backends, payment systems, and other services, allowing you to extend functionality without disrupting the core framework.
- **Design Components**: Building trust is key in ecommerce. With [Daffodil's component kit](/docs/design), streamline development and create a polished, professional store that customers will love.
- **Performance**:  With a focus on speed, Daffodil ensures a smooth and fast user experience for both customers and administrators, even as your store grows.
- **Modular Packages**: Daffodil’s architecture is based on [interoperable packages](/docs/packages), so you can select the *exact* components you need. This makes it easy to build a solution tailored specifically to your requirements.
- **Headless**: Built to integrate seamlessly with modern platforms, Daffodil serves as the frontend "head" to your headless system.
```

Author: fc1f8b9c99ed05eefb9cdfe0715182039f0646b7

> - **Drivers** [Drivers](/guides/drivers) are Daffodil's lifeblood. The drivers enable you to easily integrate with a variety of eCommerce backends, payment systems, and other services, allowing you to extend functionality without disrupting the core framework.

Reviewer 2: I think the 3rd person is the correct narration perspective for this kind of technical document. If you must specifically represent our views, you should say something like `the maintainers believe...`, a la "the authors" in a scientific paper.

`The drivers enable you to easily integrate ` should be something like `The drivers enable easy integration`

Author: I personally prefer the informal tonal difference here. I find documentation that speaks to "me" as the reader friendlier and more approachable. 

I've always found 

- https://vuejs.org/guide/introduction.html
- https://laravel.com/docs/12.x#why-laravel
- https://angular.dev/guide/signals

Much easier to process than a typical scientific paper.

> - **Drivers** [Drivers](/guides/drivers) are Daffodil's lifeblood. The drivers enable you to easily integrate with a variety of eCommerce backends, payment systems, and other services, allowing you to extend functionality without disrupting the core framework.

Reviewer: `ecommerce` instead of `eCommerce`

Author: Fixed.

> - **Drivers** [Drivers](/guides/drivers) are Daffodil's lifeblood. The drivers enable you to easily integrate with a variety of eCommerce backends, payment systems, and other services, allowing you to extend functionality without disrupting the core framework.

Reviewer: This link exists, but it should be `/docs/guides/drivers.md`

Author: Fixed.

- **Design Components**: Building trust is key in ecommerce. With [Daffodil's component kit](/docs/design), you can streamline development and create a polished, professional store that customers will love.

> - **Performance**: Performance is paramount. Your store can only be performant if our code is fast, you can rely on us to prevent and fix performance issues as quickly as possible.

Reviewer: ```suggestion
- **Performance**: Performance is paramount. Your store can only be performant if our code is fast. You can rely on us to prevent and fix performance issues as quickly as possible.
```

> - **Modular Packages**: Daffodil’s architecture is based on [interoperable packages](/docs/packages), so you can select *exactly* the components you need. This makes it easy to build a solution tailored specifically to your requirements.

Reviewer: ```suggestion
- **Modular Packages**: Daffodil’s architecture is based on [interoperable packages](/docs/packages), so you can select the *exact* components you need. This makes it easy to build a solution tailored specifically to your requirements.
```

Daffodil is built with flexibility and scalability in mind. Here are a few features that we believe make Daffodil special:

> - **Drivers**: [Drivers](/docs/guides/drivers.md) are Daffodil's lifeblood. The drivers enable you to easily integrate with a variety of ecommerce backends, payment systems, and other services, allowing you to extend functionality without disrupting the core framework.

Reviewer: The drivers enable you to easily integrate with various ecommerce backends, payment systems, and other services, allowing you to extend functionality without disrupting the core framework.

- **Design Components**: Building trust is key in ecommerce. With [Daffodil's component kit](/docs/design), you can streamline development and create a polished, professional store that customers will love.

> - **Performance**: Performance is paramount. Your store can only be performant if our code is fast, you can rely on us to prevent and fix performance issues as quickly as possible.

Reviewer: Performance is paramount. Your store can only be performant if our code is fast. You can rely on us to prevent and fix performance issues as quickly as possible.
